File: src/types/detection.ts
export interface Detection {
  x1: number;
  y1: number;
  x2: number;
  y2: number;
  confidence: number;
  class: number;
  label: string;
}

export interface DetectionResults {
  [modelName: string]: {
    detections: Detection[];
    count: number;
    error?: string;
  };
}

export interface ModelOption {
  id: string;
  name: string;
  label: string;
  color?: string;
}

export interface WebSocketMessage {
  type: 'frame' | 'detections' | 'error';
  image?: string;
  models?: string[];
  timestamp?: number;
  results?: DetectionResults;
  error?: string;
}

export type ConnectionStatus = 'disconnected' | 'connecting' | 'connected';

File: app/schemas/detection.py
from pydantic import BaseModel
from typing import List, Optional, Dict


class Detection(BaseModel):
    x1: float
    y1: float
    x2: float
    y2: float
    confidence: float
    class_id: int
    label: str


class ModelResult(BaseModel):
    detections: List[Detection]
    count: int
    model: str
    error: Optional[str] = None


class DetectionResponse(BaseModel):
    camera_id: str
    timestamp: int
    results: Dict[str, ModelResult]
    type: str = "detections"